File: common/src/main/java/com/ssblur/scriptor/trade/TierTrade.java
package com.ssblur.scriptor.trade;

import com.ssblur.scriptor.item.ScriptorItems;
import net.minecraft.util.RandomSource;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.npc.VillagerTrades;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.item.trading.ItemCost;
import net.minecraft.world.item.trading.MerchantOffer;
import org.jetbrains.annotations.Nullable;

public class TierTrade implements VillagerTrades.ItemListing {
  @Nullable
  @Override
  public MerchantOffer getOffer(Entity entity, RandomSource randomSource) {
    int tier = randomSource.nextInt(3);

    Item book;
    Item book2;
    switch (tier) {
      case 0 -> {
        book = ScriptorItems.TOME_TIER2.get();
        book2 = ScriptorItems.TOME_TIER1.get();
      }
      case 1 -> {
        book = ScriptorItems.TOME_TIER3.get();
        book2 = ScriptorItems.TOME_TIER2.get();
      }
      default -> {
        book = ScriptorItems.TOME_TIER4.get();
        book2 = ScriptorItems.TOME_TIER3.get();
      }
    }

    return new MerchantOffer(
      new ItemCost(book, 1),
      new ItemStack(book2, 2),
      1,
      15,
      0.05f
    );
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/trade/TierTrade.kt
package com.ssblur.scriptor.trade

import com.ssblur.scriptor.item.ScriptorItems.TOME_TIER1
import com.ssblur.scriptor.item.ScriptorItems.TOME_TIER2
import com.ssblur.scriptor.item.ScriptorItems.TOME_TIER3
import com.ssblur.scriptor.item.ScriptorItems.TOME_TIER4
import net.minecraft.util.RandomSource
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.npc.VillagerTrades
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.trading.ItemCost
import net.minecraft.world.item.trading.MerchantOffer

class TierTrade : VillagerTrades.ItemListing {
    override fun getOffer(entity: Entity, randomSource: RandomSource): MerchantOffer? {
        val tier = randomSource.nextInt(3)

        val book: Item
        val book2: Item
        when (tier) {
            0 -> {
                book = TOME_TIER2.get()
                book2 = TOME_TIER1.get()
            }

            1 -> {
                book = TOME_TIER3.get()
                book2 = TOME_TIER2.get()
            }

            else -> {
                book = TOME_TIER4.get()
                book2 = TOME_TIER3.get()
            }
        }
        return MerchantOffer(
            ItemCost(book, 1),
            ItemStack(book2, 2),
            1,
            15,
            0.05f
        )
    }
}
